{"cells": [{"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "TP-xE7XZQoN9", "outputId": "cb83a672-868d-48c6-eb88-210c80721b56"}, "cell_type": "code", "source": "pwd\n", "execution_count": 13, "outputs": [{"output_type": "execute_result", "execution_count": 13, "data": {"text/plain": "'/home/wsuser/work'"}, "metadata": {}}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "bI3BXjh3Q8Eh", "outputId": "0a705756-f784-4375-f536-416a908392b3"}, "cell_type": "code", "source": "cd /home/wsuser/work", "execution_count": 6, "outputs": [{"output_type": "stream", "text": "/home/wsuser/work\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "mOFzlvTURIQ3", "outputId": "d0b5a23e-0f0a-4681-e118-a5a3038250db"}, "cell_type": "code", "source": "!pip install keras\n!pip install tensorflow", "execution_count": 14, "outputs": [{"output_type": "stream", "text": "Requirement already satisfied: keras in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (2.7.0)\nRequirement already satisfied: tensorflow in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (2.7.2)\nRequirement already satisfied: numpy>=1.14.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.20.3)\nRequirement already satisfied: gast<0.5.0,>=0.2.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.4.0)\nRequirement already satisfied: protobuf>=3.9.2 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (3.19.1)\nRequirement already satisfied: opt-einsum>=2.3.2 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (3.3.0)\nRequirement already satisfied: astunparse>=1.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.6.3)\nRequirement already satisfied: tensorflow-estimator<2.8,~=2.7.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.7.0)\nRequirement already satisfied: wheel<1.0,>=0.32.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.37.0)\nRequirement already satisfied: h5py>=2.9.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (3.2.1)\nRequirement already satisfied: keras<2.8,>=2.7.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.7.0)\nRequirement already satisfied: typing-extensions>=3.6.6 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (4.1.1)\nRequirement already satisfied: flatbuffers<3.0,>=1.12 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.0)\nRequirement already satisfied: tensorboard~=2.7 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (2.7.0)\nRequirement already satisfied: termcolor>=1.1.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.1.0)\nRequirement already satisfied: keras-preprocessing>=1.1.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.1.2)\nRequirement already satisfied: absl-py>=0.4.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.12.0)\nRequirement already satisfied: grpcio<2.0,>=1.24.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.42.0)\nRequirement already satisfied: wrapt>=1.11.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.12.1)\nRequirement already satisfied: tensorflow-io-gcs-filesystem>=0.21.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.23.1)\nRequirement already satisfied: google-pasta>=0.1.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (0.2.0)\nRequirement already satisfied: six>=1.12.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorflow) (1.15.0)\nRequirement already satisfied: setuptools>=41.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (58.0.4)\nRequirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (0.6.1)\nRequirement already satisfied: tensorboard-plugin-wit>=1.6.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (1.6.0)\nRequirement already satisfied: google-auth<3,>=1.6.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (1.23.0)\nRequirement already satisfied: requests<3,>=2.21.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (2.26.0)\nRequirement already satisfied: markdown>=2.6.8 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (3.3.3)\nRequirement already satisfied: werkzeug>=0.11.15 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (2.0.2)\nRequirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from tensorboard~=2.7->tensorflow) (0.4.4)\nRequirement already satisfied: rsa<5,>=3.1.4 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (4.7.2)\nRequirement already satisfied: cachetools<5.0,>=2.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (4.2.2)\nRequirement already satisfied: pyasn1-modules>=0.2.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (0.2.8)\nRequirement already satisfied: requests-oauthlib>=0.7.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.7->tensorflow) (1.3.0)\nRequirement already satisfied: pyasn1<0.5.0,>=0.4.6 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard~=2.7->tensorflow) (0.4.8)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (2022.9.24)\nRequirement already satisfied: urllib3<1.27,>=1.21.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (1.26.7)\nRequirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (2.0.4)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests<3,>=2.21.0->tensorboard~=2.7->tensorflow) (3.3)\nRequirement already satisfied: oauthlib>=3.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard~=2.7->tensorflow) (3.2.1)\n", "name": "stdout"}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/", "height": 35}, "id": "h5sh8spkROa4", "outputId": "ebae9989-0eae-4f9e-a46d-0f863d510801"}, "cell_type": "code", "source": "pwd", "execution_count": 32, "outputs": [{"output_type": "execute_result", "execution_count": 32, "data": {"text/plain": "'/home/wsuser/work'"}, "metadata": {}}]}, {"metadata": {}, "cell_type": "code", "source": "from keras.models import Sequential\nfrom keras.layers import Dense\nfrom keras.layers import Convolution2D\nfrom keras.layers import MaxPooling2D\nfrom keras.layers import Flatten", "execution_count": 28, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "\nimport os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='ccAqZx_IM2IAQrD_b0-CuU1QCMZDzMqqYM4N7oIRB-Wu',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'naturaldisasterintensityanalysisa-donotdelete-pr-pkpliyi4wdyll3'\nobject_key = 'dataset.zip'\n\nstreaming_body_10 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n", "execution_count": 25, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "", "execution_count": null, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "0UFADofSRQYo", "outputId": "8188f557-a02c-4753-9c96-816feabf6772"}, "cell_type": "code", "source": "from io import BytesIO\nimport zipfile\nunzip= zipfile.ZipFile(BytesIO(streaming_body_10.read()),'r')\nfile_paths=unzip.namelist()\nfor path in file_paths:\n    unzip.extract(path)", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "", "execution_count": null, "outputs": []}, {"metadata": {"id": "jtRpW0RrYEs0"}, "cell_type": "code", "source": "from tensorflow.keras.preprocessing.image import ImageDataGenerator", "execution_count": 34, "outputs": []}, {"metadata": {"id": "RmMLbdHwYE7S"}, "cell_type": "code", "source": "train_datagen=ImageDataGenerator(rescale=1./255,zoom_range=0.2,horizontal_flip=True,shear_range=0.2)", "execution_count": 35, "outputs": []}, {"metadata": {"id": "2W6jGrVpYFEx"}, "cell_type": "code", "source": "test_datagen=ImageDataGenerator(rescale=1./255)", "execution_count": 36, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "_YizQBchYFWB", "outputId": "45bc60be-596a-4774-de5f-c282eeb44ff8"}, "cell_type": "code", "source": "x_train=train_datagen.flow_from_directory(r\"/home/wsuser/work/dataset/train_set\",target_size=(64,64),\n                                                            batch_size=5,color_mode='rgb',class_mode='categorical')", "execution_count": 1, "outputs": [{"output_type": "error", "ename": "NameError", "evalue": "name 'train_datagen' is not defined", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)", "\u001b[0;32m/tmp/wsuser/ipykernel_164/2587888446.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m x_train=train_datagen.flow_from_directory(r\"/home/wsuser/work/dataset/train_set\",target_size=(64,64),\n\u001b[0m\u001b[1;32m      2\u001b[0m                                                             batch_size=5,color_mode='rgb',class_mode='categorical')\n", "\u001b[0;31mNameError\u001b[0m: name 'train_datagen' is not defined"]}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "JTJEIuAXYFca", "outputId": "85ae0823-4cf8-498d-b544-694c367e6082"}, "cell_type": "code", "source": "x_test=test_datagen.flow_from_directory(r\"/home/wsuser/work/dataset/test_set\",target_size=(64,64),\n                                                            batch_size=5,color_mode='rgb',class_mode=\"categorical\")", "execution_count": 2, "outputs": [{"output_type": "error", "ename": "NameError", "evalue": "name 'test_datagen' is not defined", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)", "\u001b[0;32m/tmp/wsuser/ipykernel_164/3133615255.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m x_test=test_datagen.flow_from_directory(r\"/home/wsuser/work/dataset/test_set\",target_size=(64,64),\n\u001b[0m\u001b[1;32m      2\u001b[0m                                                             batch_size=5,color_mode='rgb',class_mode=\"categorical\")\n", "\u001b[0;31mNameError\u001b[0m: name 'test_datagen' is not defined"]}]}, {"metadata": {"id": "bFwKaaxwYFg7"}, "cell_type": "code", "source": "import numpy as np\nimport tensorflow\nfrom tensorflow.keras.models import Sequential\nfrom tensorflow.keras.layers import Dense,Conv2D,MaxPooling2D,Flatten", "execution_count": null, "outputs": []}, {"metadata": {"id": "_BzyCWCvYFlq"}, "cell_type": "code", "source": "model=Sequential()\nmodel.add(Conv2D(32,(3,3),activation=\"relu\",input_shape=(64,64,3)))\nmodel.add(MaxPooling2D(pool_size=(2,2)))\nmodel.add(Conv2D(32,(3,3),activation=\"relu\"))\nmodel.add(MaxPooling2D(pool_size=(2,2)))\nmodel.add(Flatten())\nmodel.add(Dense(units=128,activation='relu'))\nmodel.add(Dense(units=4,activation='softmax'))\nmodel.compile(loss=\"categorical_crossentropy\",metrics=[\"accuracy\"],optimizer='adam')", "execution_count": null, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "I5EEyokOYFqC", "outputId": "29d39da0-b10a-4287-f746-6a3680cea50e"}, "cell_type": "code", "source": "model.summary()", "execution_count": null, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "y63NbNX1YGED", "outputId": "6ffdabfe-1ab8-44dc-9494-4f7dce328c57"}, "cell_type": "code", "source": "model.fit_generator(generator=x_train,epochs=20,steps_per_epoch=len(x_train),validation_data=x_test,validation_steps=len(x_test))", "execution_count": null, "outputs": []}, {"metadata": {"id": "vZHPrcRudvqM"}, "cell_type": "code", "source": "model.save('disaster.h5')\nmodel_json=model.to_json()\nwith open(\"model-bw.json\",\"w\") as json_file:\n  json_file.write(model_json) ", "execution_count": null, "outputs": []}, {"metadata": {"id": "AfZNe7oCdvxy"}, "cell_type": "code", "source": "from tensorflow.keras.models import load_model\nfrom tensorflow.keras.preprocessing import image\nmodel=load_model('disaster.h5')", "execution_count": null, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "xKpbAInkdv4L", "outputId": "238c862f-143d-4eda-a3e4-13a8c43912ce"}, "cell_type": "code", "source": "x_train.class_indices", "execution_count": 3, "outputs": [{"output_type": "error", "ename": "NameError", "evalue": "name 'x_train' is not defined", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)", "\u001b[0;32m/tmp/wsuser/ipykernel_164/2819503682.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mx_train\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mclass_indices\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m", "\u001b[0;31mNameError\u001b[0m: name 'x_train' is not defined"]}]}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "lBst0L6Adv7T", "outputId": "9fd3d09d-1d22-42ef-dbec-c32d14493a65"}, "cell_type": "code", "source": "img=image.load_img(r\"/content/drive/MyDrive/Disaster/dataset/test_set/Earthquake/1329.jpg\",target_size=(64,64))\nx=image.img_to_array(img)\nx=np.expand_dims(x,axis=0)\nindex=['Cyclone','Earthquake','Flood','Wildfire']\ny=np.argmax(model.predict(x),axis=1)\nprint(index[int(y)])", "execution_count": null, "outputs": []}, {"metadata": {"colab": {"base_uri": "https://localhost:8080/"}, "id": "utB1ZvsWdv-H", "outputId": "23d4f051-8411-48c7-fb28-a19310f06539"}, "cell_type": "code", "source": "img=image.load_img(r\"/content/drive/MyDrive/Disaster/dataset/test_set/Cyclone/900.jpg\",target_size=(64,64))\nx=image.img_to_array(img)\nx=np.expand_dims(x,axis=0)\nindex=['Cyclone','Earthquake','Flood','Wildfire']\ny=np.argmax(model.predict(x),axis=1)\nprint(index[int(y)])", "execution_count": null, "outputs": []}, {"metadata": {}, "cell_type": "code", "source": "", "execution_count": null, "outputs": []}], "metadata": {"colab": {"provenance": []}, "kernelspec": {"name": "python3", "display_name": "Python 3.9", "language": "python"}, "language_info": {"name": "python", "version": "3.9.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat": 4, "nbformat_minor": 1}